src/libs/actions/PaymentMethods.ts

```diff
@@ -228,40 +236,53 @@ function addSubscriptionPaymentCard(cardData: {
     const optimisticData: OnyxUpdate[] = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM,
+            key: ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM,
             value: {isLoading: true},
         },
     ];
 
     const successData: OnyxUpdate[] = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM,
+            key: ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM,
             value: {isLoading: false},
         },
     ];
 
     const failureData: OnyxUpdate[] = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM,
+            key: ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM,
             value: {isLoading: false},
         },
     ];
 
-    // TODO integrate API for subscription card as a follow up
-    API.write(WRITE_COMMANDS.ADD_PAYMENT_CARD, parameters, {
-        optimisticData,
-        successData,
-        failureData,
-    });
+    if (currency === CONST.CURRENCY.GBP) {
+        // eslint-disable-next-line rulesdir/no-api-side-effects-method
+        API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.ADD_PAYMENT_CARD_GBR, parameters, {optimisticData, successData, failureData}).then((response) => {
+            if (response?.jsonCode !== CONST.JSON_CODE.SUCCESS) {
+                return;
+            }
+            // TODO 3ds flow will be done as a part https://github.com/Expensify/App/issues/42432
+            // We will use this onyx key to open Modal and preview iframe. Potentially we can save the whole object which come from side effect
+            Onyx.set(ONYXKEYS.VERIFY_3DS_SUBSCRIPTION, (response as {authenticationLink: string}).authenticationLink);
```

Comment:
Anyway we can avoid using the `as` cast here? Defeats the purpose of using TypeScript if we type cast often.

Reply:
Yeah - it should be covered here https://github.com/Expensify/App/pull/44321/files